<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1138</width>
    <height>693</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Scoreboard Alogorithm Simulator</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>50</y>
      <width>511</width>
      <height>221</height>
     </rect>
    </property>
    <property name="title">
     <string>Instruction Status</string>
    </property>
    <widget class="QTableWidget" name="table_instructions">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>491</width>
       <height>201</height>
      </rect>
     </property>
     <attribute name="horizontalHeaderDefaultSectionSize">
      <number>60</number>
     </attribute>
     <column>
      <property name="text">
       <string>Instruction</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>i</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>j</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>k</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Issue</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Read Operands</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Execution Complete</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Write Result</string>
      </property>
     </column>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>50</y>
      <width>611</width>
      <height>221</height>
     </rect>
    </property>
    <property name="title">
     <string>Functional unit status</string>
    </property>
    <widget class="QTableWidget" name="table_funcuints">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>611</width>
       <height>201</height>
      </rect>
     </property>
     <property name="showGrid">
      <bool>true</bool>
     </property>
     <attribute name="horizontalHeaderDefaultSectionSize">
      <number>55</number>
     </attribute>
     <column>
      <property name="text">
       <string>Time</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Name</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Busy</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Op</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Dest Fi</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Fj</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Fk</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Qj</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Qk</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Rj</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Rk</string>
      </property>
     </column>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>280</y>
      <width>1141</width>
      <height>281</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="title">
     <string>Register Status</string>
    </property>
    <widget class="QTableWidget" name="table_registers">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>1121</width>
       <height>61</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>9</pointsize>
      </font>
     </property>
     <attribute name="horizontalHeaderVisible">
      <bool>true</bool>
     </attribute>
     <attribute name="horizontalHeaderDefaultSectionSize">
      <number>40</number>
     </attribute>
    </widget>
    <widget class="QTableWidget" name="table_fregisters">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>1121</width>
       <height>61</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>9</pointsize>
      </font>
     </property>
     <attribute name="horizontalHeaderDefaultSectionSize">
      <number>40</number>
     </attribute>
    </widget>
    <widget class="QTableWidget" name="table_values">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>140</y>
       <width>1121</width>
       <height>61</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>9</pointsize>
      </font>
     </property>
     <attribute name="horizontalHeaderDefaultSectionSize">
      <number>40</number>
     </attribute>
    </widget>
    <widget class="QTableWidget" name="table_fvalues">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>200</y>
       <width>1121</width>
       <height>61</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>9</pointsize>
      </font>
     </property>
     <attribute name="horizontalHeaderDefaultSectionSize">
      <number>40</number>
     </attribute>
    </widget>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>451</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>22</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Scoreboard Alogrithm Simulator</string>
    </property>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>10</y>
      <width>703</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label_current">
       <property name="text">
        <string>Current Cycle</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_cycle">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>50</width>
         <height>20</height>
        </size>
       </property>
       <property name="baseSize">
        <size>
         <width>50</width>
         <height>50</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: rgb(186, 189, 182);
color: rgb(239, 41, 41);
</string>
       </property>
       <property name="text">
        <string>0</string>
       </property>
       <property name="scaledContents">
        <bool>false</bool>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_2">
       <property name="font">
        <font>
         <family>Consolas</family>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Cycle</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="lineedit_cycle">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="maximumSize">
        <size>
         <width>62</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_loadfile">
       <property name="text">
        <string>Load File</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_run">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Run</string>
       </property>
       <property name="flat">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_pause">
       <property name="text">
        <string>Pause</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_stop">
       <property name="text">
        <string>Stop</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_setting">
       <property name="text">
        <string>setting</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>100</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>560</y>
      <width>1131</width>
      <height>131</height>
     </rect>
    </property>
    <property name="title">
     <string>Running Informations</string>
    </property>
    <widget class="QTextBrowser" name="run_info">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>1111</width>
       <height>91</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
    </widget>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
